aerleon/lib/naming.py

```diff
@@ -877,11 +909,26 @@ def _ParseYamlNetworks(self, file_data: Dict[str, Any], file_name: str) -> None:
                     continue
 
                 if comment is None:
-                    unit.items.append(value)
+                    if network_ref:
+                        addr_unit.items.append(value)
+                        fqdn_unit.items.append(value)
+                    elif ip:
+                        addr_unit.items.append(value)
+                    else:
+                        fqdn_unit.items.append(value)
                 else:
-                    unit.items.append(f'{value} # {comment}')
+                    if network_ref:
+                        addr_unit.items.append(f'{value} # {comment}')
+                        fqdn_unit.items.append(f'{value} # {comment}')
+                    elif ip:
+                        addr_unit.items.append(f'{value} # {comment}')
+                    else:
+                        fqdn_unit.items.append(f'{value} # {comment}')
+                    
 
-                if network_ref and network_ref not in self.networks:
+                if network_ref and (
+                    network_ref not in self.networks or network_ref not in self.fqdn
```

Comment:
This logic change isn't strictly necessary since line 874 always creates an entry in self.networks for a symbol even if there is only FQDN data for that symbol.

Reply:
Done